src/main/java/com/example/housingmanagement/api/services/AssignmentService.java

```diff
@@ -54,15 +54,15 @@ public void assignSpecificOccupantToSpecificHouse(HouseRequest houseRequest, Occ
                 Optional.ofNullable(occupantRepository.findByFirstNameAndLastName(occupantRequest.getFirstName(),
                         occupantRequest.getLastName()));
         //update database entry for the identified occupant
-        OccupantInternalEntity occupantInternalEntityToBeModified =
-                new OccupantInternalEntity(
-                        occupantInternalEntityToAssign.get().getId(),
-                        occupantInternalEntityToAssign.get().getFirstName(),
-                        occupantInternalEntityToAssign.get().getLastName(),
-                        occupantInternalEntityToAssign.get().getGender(),
-                        houseInternalEntityToAssign.get());
-        occupantRepository.save(occupantInternalEntityToBeModified);
+        if (occupantInternalEntityToAssign.isPresent() && houseInternalEntityToAssign.isPresent()) {
+            OccupantInternalEntity occupantInternalEntityToBeModified =
+                    new OccupantInternalEntity(
```

Comment:
tutaj powtarza się wywołanie `occupantInternalEntityToAssign.get()` - możesz to zrobić raz wcześniej i przypisać do zmiennej. I na tej nowej zmiennej robić dalej wywołania getów
A jak to zrobisz, to potem sobie to znowu trochę poprawiamy ;)

Reply:
nice, czyli ctrl+alt+V introduce a variable